File: problems/custom/matlab_problems/custom3.m
function problem = custom3()
    % This is a toy example to show how to construct a .m file that returns
    % a instance of Problem class representing a linearly constrained problem.

    p_struct.name = 'custom2';
    p_struct.x0 = [1; 1];
    p_struct.fun = @fun;
    p_struct.xl = [-5; -5];
    p_struct.xu = [5; 5];
    p_struct.aub = [1, 1];
    p_struct.bub = 1;
    p_struct.aeq = [1, -1];
    p_struct.beq = 0;

    problem = Problem(p_struct);
end

function f = fun(x)
    f = sum(x.^2);
end

File: python/optiprofiler/problem_libs/custom/python_problems/custom3.py
def custom3():
    """
    This is a toy example to show how to construct a .py file that returns a dictionary that describes a linearly constrained optimization problem.
    """
    p_dict = {}
    p_dict['name'] = 'custom3'
    p_dict['x0'] = [1.0, 1.0]
    p_dict['fun'] = fun
    p_dict['xl'] = [-5.0, -5.0]
    p_dict['xu'] = [5.0, 5.0]
    p_dict['aub'] = [1.0, 1.0]
    p_dict['bub'] = 1
    p_dict['aeq'] = [1.0, -1.0]
    p_dict['beq'] = 0

    return p_dict


def fun(x):
    return sum(xi**2 for xi in x)